xrootd/client/session.go

```diff
@@ -0,0 +1,252 @@
+package client
+
+import (
+	"context"
+	"encoding/binary"
+	"io"
+	"log"
+	"net"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/internal/mux"
+	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+	"go-hep.org/x/hep/xrootd/xrdproto"
+	"go-hep.org/x/hep/xrootd/xrdproto/signing"
+	"go-hep.org/x/hep/xrootd/xrdproto/sigver"
+)
+
+type session struct {
```

Comment:
please document this type.
it's not exported, but it's central to the use of that package and how the client ticks.
so, it's paramount to know how it works to be able to maintain the whole package :)